google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java

```diff
@@ -295,12 +295,8 @@ private static SpannerException fromApiException(ApiException exception) {
       code = Status.Code.UNKNOWN;
     }
     ErrorCode errorCode = ErrorCode.fromGrpcStatus(Status.fromCode(code));
-    if (exception.getCause() != null) {
-      return SpannerExceptionFactory.newSpannerException(
-          errorCode, exception.getMessage(), exception.getCause());
-    } else {
-      return SpannerExceptionFactory.newSpannerException(errorCode, exception.getMessage());
-    }
+    return SpannerExceptionFactory.newSpannerException(
```

Comment:
(First of all: I think this in general seems like a good idea)

I wonder whether this modification makes this technically a breaking change. The exception that is returned here, and thereby also to users, will change slightly as the underlying cause will now be different than previously. Any code that inspects and makes any kind of choice based on the cause of an exception returned by this method could start behaving differently after this change has been merged.

Reply:
I don't have a strong opinion on this, but if you do decide it's a breaking change, please make sure the commit message is `feat!: Error Details Improvement` with an exclamation mark before merging in, so that our release bot will make a major version bump for this repo.